### ISSUE TITLE
Angular Wont Update code changes while debugging after save

### ISSUE THREAD

**Issue by blogcraft** [angular/angular-cli#26633]
### Which @angular/* package(s) are the source of the bug?

compiler-cli

### Is this a regression?

Yes

### Description

Since Angular 17 if I am debugging and app with ng serve, code changes made in my submodules trigger a recompile, browser refreshes, but the same unchanged code is still running.

### Please provide a link to a minimal reproduction of the bug

https://github.com/blogcraft/WontUpdateWhileDebugging

### Please provide the exception or error you saw

```true
There is no error or Exception, the cli just recompiles as normal and shows the usual chunk files message and the message:

Application bundle generation complete. [0.077 seconds]
```

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI: 17.0.6
Node: 20.9.0
Package Manager: npm 10.1.0
OS: win32 x64

Angular: 17.0.6
... animations, cli, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... platform-server, router, service-worker

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1700.6
@angular-devkit/build-angular   17.0.6
@angular-devkit/core            17.0.6
@angular-devkit/schematics      17.0.6
@angular/cdk                    17.0.3
@angular/material               17.0.3
@schematics/angular             17.0.6
rxjs                            7.8.1
typescript                      5.2.2
zone.js                         0.14.2
```

### Anything else?

To see the behaviour while debugging, try to change the message in the `src\app\modules\signed-out-module\signed-out.component.ts` file.

**Comment by alan-agius4** [comment]
@blogcraft, kindly make the reproduction accessible.

**Comment by blogcraft** [comment]
> @blogcraft, kindly make the reproduction accessible.
> 
> Also, which IDE are you using?

sorry, its public now.

I have tried with VS Code also with simple notepad.

**Comment by alan-agius4** [comment]
Hi @blogcraft, I downloaded the reproduction, however I am seeing the changed reflected in the browser as expected.

**Comment by blogcraft** [comment]
Starts like this:
![image](https://github.com/angular/angular-cli/assets/5615323/1a6dfda8-3313-44a9-85c5-1724bfacc3aa)

Change some code...
![image](https://github.com/angular/angular-cli/assets/5615323/11e23737-7fa6-45ae-9cc1-b57e7a4ebb0f)

Recompiles:
![image](https://github.com/angular/angular-cli/assets/5615323/a0ce52f4-ff1c-46b2-b500-0fcc2ac34d38)

Text remains the same
![image](https://github.com/angular/angular-cli/assets/5615323/a85bc96c-fbc3-4b8b-a815-73a7a2d7f7b3)

**Comment by pedrosanguineti** [comment]
It's happening to me as well. Also, I managed to reproduce the issue following the example provided by @blogcraft.

```

Angular CLI: 17.0.6
Node: 18.17.0
Package Manager: npm 9.6.7
OS: win32 x64

Angular:
...
Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.1700.6 (cli-only)
@angular-devkit/core         17.0.6 (cli-only)
@angular-devkit/schematics   17.0.6 (cli-only)
@schematics/angular          17.0.6 (cli-only)
```

**Comment by jimyhdolores** [comment]
The same thing happens to me, any update on this?

**Comment by alan-agius4** [comment]
Hi @blogcraft, 

I tried this again and I was still unable to replicate the reported behaviour using the project provided. I am using `ng serve`.

I used Edge and Chrome on Windows and Chrome on Linux and the changes were always reflected in the browser as expected.

[cast (1).webm](https://github.com/angular/angular-cli/assets/17563226/57825cb9-d9ac-4fe7-89e7-499c4261eca4)

**Comment by blogcraft** [comment]
This is my test.

https://github.com/angular/angular-cli/assets/5615323/3491a703-524f-4516-a482-b6f5563c840e

Is there anything else that could impact some environments and others not?

I've tried with bash, cmd, Powershell.. Node 18, node 20.9

**Comment by alan-agius4** [comment]
@blogcraft, can you try in an incognito browser?

**Comment by blogcraft** [comment]
Same behaviour with incognito. 

My OS is in Spanish, I don't know if it has anything to do, but at this time I don't know what else to look.

**Comment by blogcraft** [comment]
Also checked If I had any global package in npm that could be doing something, but nothing outside of CLI.

![image](https://github.com/angular/angular-cli/assets/5615323/e92ecb4c-0a88-4456-b2ac-1f6acface3ad)

**Comment by alan-agius4** [comment]
@blogcraft, I am not quite sure why you are experiencing this issue. @clydin also tried your reproduction and was not able to reproduce.

**Comment by victorpd07** [comment]
Same issue, changes are not reflected after recompiling.
![image](https://github.com/angular/angular-cli/assets/153747894/38821312-4522-41d6-a736-86b935446406)

**Comment by Explodingcat** [comment]
same for me, i try in edge and chrome in an incognito

https://github.com/angular/angular-cli/assets/11544897/12d17d61-91fc-4310-ab7e-212dcabd69c6

**Comment by flonussi** [comment]
> Happened to me too. In my case it was not consistant or linked to a single file, it just happened randomly once or twice every hour or so. Shutting down the ng serve and restarting it fixed the problem.

Same here.
My whole team has this issue, about 5 times per hour

**Comment by alan-agius4** [comment]
@blogcraft can you try using the prerelease version of Angular CLI, as under the hood this uses Vite 5. 

```
ng update @angular/cli@next
```

**Comment by 32ouik7jmn** [comment]
I am also facing this problem and the problem is that the issue occurs **randomly**.

I start the server, make the change and the change is visible, repeat this X times everything works fine.
I restart the server and make a change and the change is not visible.
I restart the server again, make a change and the change is visible, repeat X times, everything works fine, only to stop working again after x+1 times.

I also tested with `aot:false` but it doesn't help

```
ng new --routing=true --style=scss --ssr=false
```

**Comment by blogcraft** [comment]
> @blogcraft can you try using the prerelease version of Angular CLI, as under the hood this uses Vite 5.
> 
> ```
> ng update @angular/cli@next
> ```
It threw an error with an incompatible dependency `@angular-eslint/schematics` 

But forced the update with `ng update @angular/cli@next --force`

![image](https://github.com/angular/angular-cli/assets/5615323/4b99bf84-ce35-4fdb-9191-0732afc5fbf9)

And so far, it seems to be working fine with `@angular/cli@next` 🥳

I'll do some further testing in the full app to see if the problem repeats itself.

**Comment by alan-agius4** [comment]
Glad to hear that the prerelease version fixes the issue @blogcraft.

**Comment by edumserrano** [comment]
@alan-agius4 I'd like to add here that I've also faced a similar issue. I thought it was related with another tool at first and I described the problem in detail here [microsoft/playwright#28632].

For me, the problem wasn't that the app didn't rebuild when debugging, it was that the app would stop rebuilding when there were several files generated in a directory that was being watched. 

In this issue [microsoft/playwright#28632] I provide a sample app which is really just the template we get from doing `ng new` using Angular `17.0.6`. The only additions to the app from the `ng new` template is that I added a button using Angular material and I added `Playwright tests`.

It should be easy to replicate the issue from my demo but let me know if you need more info.

Lastly, I did try your suggestion to use the prerelease version of Angular CLI by doing:
```
ng update @angular/cli@next
```

**And with that, using Angular `17.1.0-next.2` the issue seems to be resolved. I can't reproduce the bug anymore.** 🎉 

I would also like to add that using the demo app from [microsoft/playwright#28632], **BEFORE** updating to `@angular/cli@next`, **ANY** fileI add/delete to the `./tests` folder was triggering an app rebuild and **AFTER the update that no longer happens**. I think this is somehow related with the bug I'm facing because when the tests fail on my demo app there are several files that are generated in the `./tests/test-results` directory and that somehow seems to break the file watcher. 

@alan-agius4  I believe the problem I'm facing is related with this issue but let me know if you prefer that I open a new one.

**Comment by alan-agius4** [comment]
@edumserrano, for the description it looks like your issue is unrelated to this. This issue is about updates not being visible in the browser. Therefore,  it would be best to create a new issue with a minimal reproduction. Thanks.

**Comment by edumserrano** [comment]
@alan-agius4 note that the updates are also not visible in the browser because the app is not being rebuilt...

**Comment by alan-agius4** [comment]
@edumserrano, yes please as you are describing another issue. Although from your comment above, I get that the rebuild issue has been fixed in 17.1.0-next.
